Web Wallet rewording (#35)
credential-handler/chapi.io: developers/wallets/index.md

* [Working with `QueryByExample` format requests](querybyexample)
- **Example Code**: the [chapi-demo-wallet](https://github.com/credential-handler/chapi-demo-wallet) contains a full example implementation and is referenced throughout this guide.
- **Polyfill Library**: The [credential-handler-polyfill](https://github.com/credential-handler/credential-handler-polyfill) library provides the needed `CredentialHandler` API within the browser.
- **Helper Library**: The [web-credential-handler](https://github.com/credential-handler/web-credential-handler) library provides helper functions for CHAPI integration in your code.

## Web Wallet Registration

All Wallets need to be registered as a Credential Handler in the user's browser in order to store or retrieve Verifiable Credentials (VCs) from across the Web.

A Web Wallet is typically made up of three main parts:

> * `/index.html` - main page for the wallet
> * `/manifest.json` - Web App Manifest containting credential handler affordances
> * `/wallet-worker.html` - routing endpoint for passing URLs between CHAPI and the Wallet

Reviewer: To make it more clear which of these specifically named relative URLs *have* to be used and which are examples:

```suggestion
* A main page for the wallet (e.g., `/index.html`)
* Web App Manifest containing credential handler affordances (using `/manifest.json`)
* A routing endpoint for handling requests between CHAPI and the Wallet (e.g., `/wallet-worker.html`) 
```

Author: I rather liked the filename first approach as it gave me ready handles to hook onto later, but happy to swap them. I'll muse on it a bit.

Author: k...you talked me into it... 😉

### 1. Add a `credential_handler` to your Web app's `manifest.json`

> In order to register a credential handler, your Web app must serve a `manifest.json` file from its root path (`/manifest.json`). This file _must_ be CORS-enabled. At a minimum, the `manifest.json` file should contain the following `credential_handler` object:

Reviewer: We should add which CORS headers are required to help devs.

Author: @dlongley are their suggested ones? Likely don't want everyone taking the "easy" road and just `Access-Control-Allow-Origin: *` ...but we also don't want them having to chase the domain name around--should it change.

@brianorwhatever you've done this recently. Any suggestions here on CORS specifics?

Reviewer 2: * is required as it could be any domain on the web that needs to pull this file down

  target="_blank" rel="noopener noreferrer"> chapi-demo-wallet/manifest.json </a>
</p>

> The `/wallet-worker.html` page will setup listeners for CHAPI-sent events

Reviewer: ```suggestion
The `/wallet-worker.html` page (note: you can use another name if you'd like) will setup listeners for CHAPI-sent events
```

Author: I did have something like that at one point...I'll put it back.

<script>
  document.getElementById('installHandlerButton').addEventListener('click', async function() {
    try {
      await credentialHandlerPolyfill.loadOnce();
      await WebCredentialHandler.installHandler();
      console.log('Handler Installed Successfully!');
    } catch (error) {
      console.error('Error installing handler: ' + error.message);
    }
  });
</script>
```

**Alternatively**, if you're developing your Web Wallet using Node.js tooling and bundling your
applications JavaScript, you can add the [credential-handler-polyfill](https://www.npmjs.com/package/credential-handler-polyfill) library to your project via NPM. The [web-credential-handler](https://www.npmjs.com/package/web-credential-handler) helper library can also be added to simplify your code.

```sh
npm i credential-handler-polyfill@3
npm i web-credential-handler@2
```

In your code, you can import and load the polyfill library as follows:

```javascript
import * as CredentialHandlerPolyfill from 'credential-handler-polyfill';
import * as WebCredentialHandler from 'web-credential-handler';
```

Once you've done that and bundled it, reference the output/bundled `.js` in

> your `index.html`.

Reviewer: ```suggestion
your `index.html` (note: you can use another name if you'd like).
```
